### PR TITLE
[CCI] Fix EUI/OUI type errors

### LINKED ISSUE #3028: Fix EUI/OUI related type errors
Task for https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1660

```
$tsc --noEmit
...

src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx:39:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

39 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/advanced_settings/public/management_app/components/form/form.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/advanced_settings/public/management_app/components/search/search.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/advanced_settings/public/management_app/components/search/search.tsx:114:51 - error TS2322: Type '({ query, error }: { query: Query | null; error: {    message: string;} | null; }) => void' is not assignable to type '(args: ArgsWithQuery | ArgsWithError) => boolean | void'.
  Types of parameters '__0' and 'args' are incompatible.
    Type 'ArgsWithQuery | ArgsWithError' is not assignable to type '{ query: Query | null; error: { message: string; } | null; }'.
      Type 'ArgsWithQuery' is not assignable to type '{ query: Query | null; error: { message: string; } | null; }'.
        The types of 'query.ast' are incompatible between these types.
          Type 'import("@opensearch-project/oui/src/eui_components/search_bar/query/ast")._AST' is not assignable to type 'import("@opensearch-project/oui/src/components/search_bar/query/ast")._AST'.
            Types have separate declarations of a private property '_clauses'.

114         <EuiSearchBar box={box} filters={filters} onChange={this.onChange} query={query} />
                                                      ~~~~~~~~

  node_modules/@elastic/eui/oui.d.ts:16441:6
    16441      onChange?: (args: ArgsWithQuery | ArgsWithError) => void | boolean;
               ~~~~~~~~
    The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<EuiSearchBar> & Readonly<EuiSearchBarProps> & Readonly<...>'

src/plugins/advanced_settings/public/management_app/components/search/search.tsx:114:76 - error TS2322: Type 'Query' is not assignable to type 'QueryType | undefined'.
  Type 'import("@opensearch-project/oui/src/components/search_bar/query/query").Query' is not assignable to type 'import("@opensearch-project/oui/src/eui_components/search_bar/query/query").Query'.
    Types of property 'ast' are incompatible.
      Type 'import("@opensearch-project/oui/src/components/search_bar/query/ast")._AST' is not assignable to type 'import("@opensearch-project/oui/src/eui_components/search_bar/query/ast")._AST'.
        Types have separate declarations of a private property '_clauses'.

114         <EuiSearchBar box={box} filters={filters} onChange={this.onChange} query={query} />
                                                                               ~~~~~

  node_modules/@elastic/eui/oui.d.ts:16449:6
    16449      query?: QueryType;
               ~~~~~
    The expected type comes from property 'query' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<EuiSearchBar> & Readonly<EuiSearchBarProps> & Readonly<...>'

src/plugins/dashboard/public/application/dashboard_empty_screen.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

31 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/dashboard/public/application/tests/dashboard_container.test.tsx:31:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

31 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/data/public/ui/shard_failure_modal/shard_failure_open_modal_button.test.tsx:37:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

37 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/dev_tools/public/plugin.ts:74:7 - error TS2322: Type '"/plugins/home/assets/logos/opensearch_mark_default.svg"' is not assignable to type '"string" | "number" | "function" | "color" | "copy" | "grab" | "help" | "grid" | "invert" | "dot" | "menu" | "filter" | "scale" | "wordWrap" | "cross" | "push" | "link" | "image" | ... 405 more ... | undefined'.

74       euiIconType: '/plugins/home/assets/logos/opensearch_mark_default.svg',
         ~~~~~~~~~~~

  src/core/target/types/public/application/types.d.ts:150:5
    150     euiIconType?: EuiIconType;
            ~~~~~~~~~~~
    The expected type comes from property 'euiIconType' which is declared here on type 'App<unknown>'

src/plugins/discover/public/application/angular/context/components/action_bar/action_bar.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/angular/doc_table/components/pager/tool_bar_pager_buttons.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/angular/doc_table/components/table_header/table_header.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/context_error_message/context_error_message.test.tsx:37:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

37 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/doc/doc.test.tsx:36:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

36 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/doc_viewer/doc_viewer.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/hits_counter/hits_counter.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/loading_spinner/loading_spinner.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/sidebar/discover_field_search.test.tsx:34:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

34 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/sidebar/discover_sidebar.test.tsx:33:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

33 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/table/table.test.tsx:33:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

33 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/discover/public/application/components/timechart_header/timechart_header.test.tsx:36:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

36 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/embeddable/public/lib/embeddables/embeddable_root.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/embeddable/public/lib/panel/embeddable_panel.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/add_panel_flyout.test.tsx:44:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

44 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/input_control_vis/public/components/editor/controls_tab.test.tsx:33:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

33 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/input_control_vis/public/components/editor/list_control_editor.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/input_control_vis/public/components/editor/range_control_editor.test.tsx:36:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

36 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/input_control_vis/public/components/vis/input_control_vis.test.tsx:35:33 - error TS2306: File '/workspaces/OpenSearch-Dashboards/node_modules/@elastic/eui/lib/test/index.d.ts' is not a module.

35 import { findTestSubject } from '@elastic/eui/lib/test';
                                   ~~~~~~~~~~~~~~~~~~~~~~~



```

### PR DESCRIPTION
**Pull request by Nicksqain**
### Description

- [x] Update  imports
- [x] fix EuiIconType type error
- [x] fix query type error in EuiSearchBar

### Issues Resolved

- closes #3028 

### Check List

- [x] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Nicksqain** [comment]
I initialised the PR to keep working on it! 
I want to know if the imports will work correctly, would you be kind enough to give feedback?

**Comment by Nicksqain** [comment]
@joshuarrrr @ashwin-pc 
I fixed the non-working imports that were declared in elastic ui, accidentally came across a folder with findTestSubject, which is identical and does not cause errors.
But what did I do wrong? I can't run snapshot
изображение

**Comment by joshuarrrr** [comment]
@Nicksqain It's not your fault or problem - it appears there's a problem with the windows snapshot images.

**Comment by joshuarrrr** [comment]
@Nicksqain The snapshot should be working again, I restarted the test automation.

**Comment by joshuarrrr** [comment]
When you get a chance, can you add an entry to the CHANGELOG.md under the "Refactoring" section?

**Comment by manasvinibs** [comment]
@Nicksqain Were you able to get the snapshot running? Also, can you add it to description on how did you test these changes?

**Comment by Nicksqain** [comment]
@manasvinibs 
Thank you for your concern! I was able to run the snapshot! Working on that issue at the moment. It's not very clear to me, but it's interesting to figure it out.

Tests? If you mean imports of tests, I've been testing them like this
yarn test:jest src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx

изображение

**Comment by joshuarrrr** [comment]
@Nicksqain Also please update the changelog, when you get the chance.

File: src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ import { FieldSetting } from '../../types';
 import { UiSettingsType, StringValidation } from '../../../../../../core/public';
 import { notificationServiceMock, docLinksServiceMock } from '../../../../../../core/public/mocks';
 
-import { findTestSubject } from '@elastic/eui/lib/test';
+import { findTestSubject } from 'test_utils/helpers';
```
Comment (ashwin-pc) [inline src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx:39]: Nice find! yep this function is equivalent to the eui one.

**Review by ashwin-pc — COMMENTED** [review]
@Nicksqain thanks for updating the changelog. I see that you have 3 tasks listed in this PR but have only completed one of them. Are you planning on completing all three in the same PR?

**Comment by Nicksqain** [comment]
@ashwin-pc I have updated the tasks. I have added changes to the entire PR. Please check, because I'm not sure if it's the way to use Query import. But this import is declared in eui

File: src/plugins/advanced_settings/public/management_app/components/search/search.tsx [hunk 2]
```diff
@@ -30,8 +30,8 @@
 
 import React, { Fragment, PureComponent } from 'react';
 import { i18n } from '@osd/i18n';
-import { EuiSearchBar, EuiFormErrorText, Query } from '@elastic/eui';
-
+import { EuiSearchBar, EuiFormErrorText } from '@elastic/eui';
+import { Query } from '@opensearch-project/oui/src/eui_components/search_bar/query';
```
Comment (ashwin-pc) [inline src/plugins/advanced_settings/public/management_app/components/search/search.tsx:34]: This change add a type error in the parent component that uses `Search`. I suggest that you instead add a `ts-ignore` comment to the ts error on line 114. e.g.

```ts
        {/* @ts-ignore The Query types that typescript complains about here are identical and is a false flag. Once OUI migration is complete, this ignore can be removed */}
        <EuiSearchBar box={box} filters={filters} onChange={this.onChange} query={query} />

```

**Review by ashwin-pc — COMMENTED** [review]
Thanks for updating the PR @Nicksqain. The rest of the changes look good to me, just one change needed

File: src/plugins/advanced_settings/public/management_app/components/search/search.tsx (hunk 2 above)
Comment (Nicksqain) [reply src/plugins/advanced_settings/public/management_app/components/search/search.tsx:34]: @ashwin-pc  That is, do not need to update the import? Will the Query error be fixed in the future?

Comment (ashwin-pc) [reply src/plugins/advanced_settings/public/management_app/components/search/search.tsx]: Yeah, we will replace all eui references to oui. That should fix it :)

**Review by BSFishy — APPROVED** [review]
+1 for removing references to `@elastic/eui/lib/test`. This will make migrations from OUI v1 to v2 much easier

**Review by ashwin-pc — APPROVED** [review]
Nice to see this finally over the finish line. @Nicksqain Tanks for sticking with it  🎉

**Review by manasvinibs — APPROVED** [review]
LGTM!

**Pull request merged**